phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -2905,14 +2918,18 @@ public void upgradeSystemTables(final String url, final Properties props) throws
                     }
                 } finally {
                     if (acquiredMutexLock) {
-                        releaseUpgradeMutex(mutexRowKey);
+                        try {
+                            releaseUpgradeMutex(mutexRowKey);
+                        } catch (IOException e) {
+                            logger.warn("Release of upgrade mutex failed ", e);
```

Comment:
Is it ok to continue on if we fail to release the upgrade mutex, or should we just let the exception be thrown?

Reply:
I think it is fine not to throw the exception because we have a TTL on the table. Typically it should not matter to the client as such. Moreover, similar behavior is also defined the code that Upgrades the SYSCAT table.